### PR TITLE
update: adding additional doc formatting

### PR DESCRIPTION
**Pull request by krneta**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-containers/blob/main/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-containers/blob/main/CONTRIBUTING.md#commit-your-change)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-containers/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-containers/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit_tests/braket/flake8_plugins/example_files/no_error_functions.py [hunk 1]
```diff
@@ -80,6 +80,41 @@ def function_9(*my_args, **my_kwargs) -> np.ndarray:
...
+
+def function_11(arg0: int, arg1: int, arg2: int) -> int:
+    """This is a description.
+    Args:
+        arg0(int): this is an arg
+
+          - this is a list of info about this arg
+            with some wrap around
+          - and it ends here
+        arg1(int): this is another arg
```
Comment (virajvchaudhari) [inline test/unit_tests/braket/flake8_plugins/example_files/no_error_functions.py:102]: Should we allow a new line here or that case also works?

File: src/braket/flake8_plugins/braket_checkstyle_plugin.py [hunk 2]
```diff
@@ -53,9 +54,9 @@ class _Visitor(ast.NodeVisitor):
     MISC_REGEX = re.compile(
         r"^(\s*)(Throws|Raises|See Also|Note|Example|Examples|Warnings)\s*:\s*$"
     )
-    ARG_INFO_REGEX = re.compile(r"^(\s*)(\*{0,2}\w*)\s*(\([^:]*\))?\s*:\s*(.*)")
+    ARG_INFO_REGEX = re.compile(r"^(\s*)`{0,2}(\*{0,2}\w*)`{0,2}\s*(\([^:]*\))?\s*:\s*(.*)")
```
Comment (virajvchaudhari) [inline src/braket/flake8_plugins/braket_checkstyle_plugin.py:57]: This is what we have currently`re.compile(r"^(\s*)(`?\*{0,2}\w*`?)\s*(\([^:]*\))?\s*:\s*(.*)")`, does your change affects the grouping from what we have currently in the logic [here](https://github.com/aws/amazon-braket-build-tools/blob/main/src/braket/flake8_plugins/braket_checkstyle_plugin.py#L190-L193)?

Comment (krneta) [reply src/braket/flake8_plugins/braket_checkstyle_plugin.py:57]: I'm not sure what you mean by the first part of it, but the new additions are not captured so they shouldn't affect the captured groups. Also, the unit tests are passing.

Comment (krneta) [reply src/braket/flake8_plugins/braket_checkstyle_plugin.py:57]: My mistake - I didn't pull before I made the branch. However, I don't think it's correct to include the ` mark in the capture and then code for removing it. I've modified the regex.

**Review by virajvchaudhari — APPROVED** [review]

**Pull request merged**